controllers/progressivesync_controller.go

```diff
@@ -267,6 +289,41 @@ func (r *ProgressiveSyncReconciler) getClustersFromSelector(ctx context.Context,
 	return secrets, nil
 }
 
+// calculateHashValue returns a hash value string for the ApplicationSet spec
+func (r *ProgressiveSyncReconciler) calculateHashedSpec(ctx context.Context, ps *syncv1alpha1.ProgressiveSync) (string, error) {
+	key := client.ObjectKeyFromObject(ps)
+	latest := syncv1alpha1.ProgressiveSync{}
+	if err := r.Client.Get(ctx, key, &latest); err != nil {
+		r.Log.Error(err, "Failed to retrieve the ProgressiveSync object while calculating the hash value")
+		return "", err
+	}
+
+	apiMetadata := strings.Split(*latest.Spec.SourceRef.APIGroup, "/")
+	genericObj := unstructured.Unstructured{}
+	genericObj.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   apiMetadata[0],
+		Kind:    latest.Spec.SourceRef.Kind,
+		Version: apiMetadata[1],
+	})
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: "argocd", Name: latest.Spec.SourceRef.Name}, &genericObj); err != nil {
```

Comment:
We need to changed the schema of the ProgressiveSync to also include the namespace inside the SourceRef. This way we will avoid the literal "argocd".

Reply:
I think this should come from https://github.com/Skyscanner/applicationset-progressive-sync/blob/main/main.go#L55 rather than changing the SourceRef.

Reply:
We now use the value of the flag. I have also updated the tests 🙂